pcl: checksum new versions

aumuell opened the pull request:
1.11.1 does not build on macos, but 1.13.1 does

aumuell committed:
pcl: checksum new versions

1.11.1 does not build on macos, but 1.13.1 does

alecbcs commented:
> 1.11.1 does not build on macos, but 1.13.1 does

@aumuell do you know if this is due to new work to support MacOS or a bug in the v1.11.1 release? Perhaps we should add a `conflicts` statement to alert the concretizer of this incompatibility.

aumuell committed:
pcl: backport clang build fix

makes all checksummed versions build on macos
closes #30653

tldahlgren reviewed (dismissed):
Confirmed the 4 new version sha256.

aumuell commented:
there was already a bug report for the build problems of pcl < 1.13 with a link to a fix: #30653

aumuell commented:
@spackbot fix style

aumuell committed:
[@spackbot] updating style on behalf of aumuell

haampie commented on var/spack/repos/builtin/packages/pcl/package.py:
```diff
@@ -14,13 +14,24 @@ class Pcl(CMakePackage):
+    version("1.12.0", sha256="606a2d5c7af304791731d6b8ea79365bc8f2cd75908006484d71ecee01d9b51c")
     version("1.11.1", sha256="19d1a0bee2bc153de47c05da54fc6feb23393f306ab2dea2e25419654000336e")
 
     depends_on("cmake@3.5:", type="build")
```
Need to bump cmake: https://github.com/PointCloudLibrary/pcl/commit/bb24d8e487842de9597e16392caeae6491ee85a0

aumuell replied:
Thank you for the very thorough review! And even digging up all the relevant changes!

haampie commented on var/spack/repos/builtin/packages/pcl/package.py:
```diff
@@ -14,13 +14,24 @@ class Pcl(CMakePackage):
     version("1.11.1", sha256="19d1a0bee2bc153de47c05da54fc6feb23393f306ab2dea2e25419654000336e")
 
     depends_on("cmake@3.5:", type="build")
     depends_on("eigen@3.1:")
```
And eigen: https://github.com/PointCloudLibrary/pcl/commit/287783791b4654d3080059cc4cc6fe652f918c2b

haampie commented on var/spack/repos/builtin/packages/pcl/package.py:
```diff
@@ -14,13 +14,24 @@ class Pcl(CMakePackage):
 
     depends_on("cmake@3.5:", type="build")
     depends_on("eigen@3.1:")
     depends_on("flann@1.7:")
```
And flann: https://github.com/PointCloudLibrary/pcl/commit/89fe1c7e82a520fb7b66b90fc827b04f25c855f2

haampie commented on var/spack/repos/builtin/packages/pcl/package.py:
```diff
@@ -14,13 +14,24 @@ class Pcl(CMakePackage):
     depends_on("cmake@3.5:", type="build")
     depends_on("eigen@3.1:")
     depends_on("flann@1.7:")
     depends_on("boost@1.55:+filesystem+date_time+iostreams+system")
```
And boost: https://github.com/PointCloudLibrary/pcl/commit/52aed361e13787ac9157da20bc54ef39277bb0c4

aumuell committed:
pcl: fix dependencies for new versions

as pointed out in review, dependencies have been updated based on information from
CHANGES.md and observing changes to CMakeLists.txt and cmake/pcl_find_boost.cmake

alecbcs commented:
ping @haampie

alecbcs reviewed (approved):
Confirmed SHAs. @aumuell thanks for the new package versions.
